docs/DeveloperGuide.md

```diff
@@ -1,38 +1,430 @@
+</div>
+
+### Architecture
+<img src="images/ArchitectureDiagram.png" alt=""/>
+
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main components of the architecture**
+
+[**`StockPal`**](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/StockPal.java) is in charge of the app launch and shut down.
+
+The bulk of the app's work is done by the following five components:
+
+* [**`UI`**](#ui-component): The UI of the App.
+* [**`Parser`**](#parser-component): Parses user input into respective commands.
+* [**`Commands`**](#commands-component): The command executor.
+* [**`Data`**](#data-component): Holds the data of the App in memory.
+* [**`Storage`**](#storage-component): Reads data from, and writes data to, the hard disk.
+
+[**`Commons`**](#common-classes) represents a collection of classes used by multiple components above.
+[**`Exceptions`**](#exceptions-classes) represents a collection of exceptions used by multiple components above.
+
+**How the architecture components interact with each other**
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+
+<img src="images/ArchitectureSequenceDiagram.png" alt=""/>
+
+The sections below give more details of each component.
+
+### UI component
+The **API** of this component is specified in [`Ui.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/tree/master/src/main/java/seedu/stockpal/ui/Ui.java)
+
+![Structure of the UI Component](images/UiClassDiagram.png)
+
+### Parser component
+
+**API** : [`Parser.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/parser/Parser.java)
+
+![Structure of the Parser Component](images/ParserClassDiagram.png)
+
+How the parsing works:
+1. When user inputs, the input is passed to the `Parser`.
+2. `Parser` first extracts the command.
+3. Using the extracted command, `Parser` will perform different validation checks on the arguments supplied in the
+input.
+4. Arguments (mainly the `pid`, `name`, `quantity`, `price`, `description`, `amount` fields) are validated. 
+Exceptions are thrown when the fields do not pass their respective type checks.
+5. Once validation passes, `Parser` uses the validated arguments to creates an instance of that particular command. 
+For example, a `delete` command will cause `Parser` to create a new instance of `DeleteCommand(pid)`.
+6. The created command object is returned back to `main` function for further processing.
+
+### Commands component
+
+**API** : [`Command.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/commands/Command.java)
+
+
+![Structure of the Commands Component](images/CommandsClassDiagram.png)
```

Comment:
It seems like the image path was not specified correctly
![Uploading Screenshot 2024-04-03 at 12.44.24.png…]()